external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaDataConsumer.scala

```diff
@@ -266,12 +293,9 @@ private[kafka010] case class InternalKafkaConsumer(
         }
       } catch {
         case e: OffsetOutOfRangeException =>
-          // When there is some error thrown, it's better to use a new consumer to drop all cached
-          // states in the old consumer. We don't need to worry about the performance because this
-          // is not a common path.
-          resetConsumer()
```

Comment:
Don't we have to use a new consumer to keep the original behavior?

Reply:
OffsetOutOfRangeException doesn't seem to indicate consumer itself is corrupted so I feel consumer is not strictly necessary to be reset, but as we don't know the case of delegation token it would be safer to get new consumer. I'll address it.

Reply:
> I feel consumer is not strictly necessary to be reset

Same understanding but if we can't give explanation why we remove functionality it would be better to keep.